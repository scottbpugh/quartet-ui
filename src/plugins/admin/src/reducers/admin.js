export const initialData = () => {
  return {};
};
export default {};
