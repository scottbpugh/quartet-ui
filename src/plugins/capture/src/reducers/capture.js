// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {handleActions} from "redux-actions";
import {showMessage} from "lib/message";
import {pluginRegistry} from "plugins/pluginRegistration";
import actions from "../actions/capture";
import serverActions from "actions/serversettings";

export const initialData = () => ({
  servers: {}
});

export const loadRules = server => {
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .getClient()
      .then(client => {
        client.apis.capture.capture_rules_list().then(result => {
          // load steps, all steps for all rules.
          // This may become an issue in the future, if so, a new backend API endpoint
          // needs to be added.
          client.apis.capture.capture_steps_list().then(steps => {
            let rules = result.body.map(rule => {
              // add steps to the rule.
              rule.steps = steps.body.filter(step => {
                if (step.rule === rule.id) {
                  return true;
                }
                return false;
              });
              return rule;
            });
            dispatch({
              type: actions.loadRules,
              payload: {
                serverID: server.serverID,
                rules: result.body
              }
            });
          });
        });
      });
  };
};

export const deleteRule = (server, rule) => {
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .getClient()
      .then(client => {
        client.apis.capture.capture_rules_delete(rule).then(result => {
          return dispatch(loadRules(server));
        });
      });
  };
};

export const loadTasks = server => {
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .getClient()
      .then(client => {
        client.apis.capture.capture_tasks_list().then(result => {
          dispatch({
            type: actions.loadTasks,
            payload: {
              serverID: server.serverID,
              tasks: result.body
            }
          });
        });
      });
  };
};

export const deleteStep = (server, step) => {
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .getClient()
      .then(client => {
        client.apis.capture.capture_steps_delete(step).then(result => {
          return dispatch(loadRules(server));
        });
      });
  };
};

export default handleActions(
  {
    [actions.loadRules]: (state, action) => {
      if (!state.servers) {
        // temporary debug fix.
        state.servers = {};
      }
      return {
        ...state,
        servers: {
          ...state.servers,
          [action.payload.serverID]: {
            ...state.servers[action.payload.serverID],
            rules: action.payload.rules
          }
        }
      };
    },
    [actions.loadTasks]: (state, action) => {
      if (!state.servers) {
        // temporary debug fix.
        state.servers = {};
      }
      return {
        ...state,
        servers: {
          ...state.servers,
          [action.payload.serverID]: {
            ...state.servers[action.payload.serverID],
            tasks: action.payload.tasks
          }
        }
      };
    },
    [serverActions.serverUpdated]: (state, action) => {
      // we want to reload pools when new server is saved.
      /*action.asyncDispatch(
        loadPools(pluginRegistry.getServer(action.payload))
      );*/
      return {
        ...state,
        servers: {
          ...state.servers,
          [action.payload.serverID]: {}
        }
      };
    }
  },
  {}
);
