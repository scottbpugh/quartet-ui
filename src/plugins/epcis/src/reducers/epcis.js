// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
import {handleActions} from "redux-actions";
import {pluginRegistry} from "plugins/pluginRegistration";
import {showMessage} from "lib/message";
import {setServerState} from "lib/reducer-helper";
import actions from "../actions/epcis";

export const initialData = () => {
  return {
    servers: {}
  };
};

export const loadEvent = (server, eventID) => {
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .fetchObject("epcis_event_detail_read", {event_id: eventID})
      .then(event => {
        return dispatch({
          type: actions.loadItemDetail,
          payload: {
            serverID: server.serverID,
            itemID: eventID,
            itemDetail: event
          }
        });
      })
      .catch(e => {
        showMessage({
          type: "error",
          id: "plugins.epcis.errorLoadingEvent",
          values: {error: e}
        });
      });
  };
};

export const loadEntry = (server, entryID) => {
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .fetchObject("epcis_events_by_entry_id_read", {entry_identifier: entryID})
      .then(entry => {
        return dispatch({
          type: actions.loadItemDetail,
          payload: {
            serverID: server.serverID,
            itemID: entryID,
            itemDetail: entry
          }
        });
      })
      .catch(e => {
        showMessage({
          type: "error",
          id: "plugins.epcis.errorLoadingEntry",
          values: {error: e}
        });
      });
  };
};

export const loadEntries = (server, search, page, ordering) => {
  const params = {};
  if (search) {
    params.search = search;
  }
  if (page) {
    params.page = page;
  }
  if (ordering) {
    params.ordering = ordering;
  }
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .fetchPageList("epcis_entries_list", params, [])
      .then(response => {
        return dispatch({
          type: actions.loadEntries,
          payload: {
            serverID: server.serverID,
            entries: response.results,
            count: response.count,
            next: response.next
          }
        });
      })
      .catch(e => {
        showMessage({
          type: "error",
          id: "plugins.epcis.errorLoadingEvents",
          values: {error: e}
        });
      });
  };
};

export const getGeoForEntry = (server, epc) => {
  return async dispatch => {
    try {
      // first remove existing events.
      dispatch({
        type: actions.clearGeoEvents,
        payload: {serverID: server.serverID}
      });
      const client = await pluginRegistry.getServer(server.serverID).getClient();
      const response = await client.apis.masterdata.masterdata_entry_geohistory_by_epc_read(
        {epc}
      );
      if (response.ok) {
        return dispatch({
          type: actions.loadGeoEvents,
          payload: {
            serverID: server.serverID,
            geoEvents: response.body
          }
        });
      }
    } catch (e) {
      showMessage({
        type: "error",
        id: "plugins.epcis.errorLoadingEntryGeo",
        values: {error: e}
      });
    }
  };
};

export const loadEvents = (server, search, page, ordering, type) => {
  const params = {};
  if (type) {
    params.type = type;
  }
  if (search) {
    params.search = search;
  }
  if (page) {
    params.page = page;
  }
  if (ordering) {
    params.ordering = ordering;
  }
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .fetchPageList("epcis_events_list", params, [])
      .then(response => {
        return dispatch({
          type: actions.loadEvents,
          payload: {
            serverID: server.serverID,
            events: response.results,
            count: response.count,
            next: response.next
          }
        });
      })
      .catch(e => {
        showMessage({
          type: "error",
          id: "plugins.epcis.errorLoadingEvents",
          values: {error: e}
        });
      });
  };
};

export default handleActions(
  {
    [actions.loadEntries]: (state, action) => {
      return setServerState(state, action.payload.serverID, {
        entries: action.payload.entries,
        count: action.payload.count,
        next: action.payload.next
      });
    },
    [actions.loadEvents]: (state, action) => {
      return setServerState(state, action.payload.serverID, {
        events: action.payload.events,
        count: action.payload.count,
        next: action.payload.next
      });
    },
    [actions.loadItemDetail]: (state, action) => {
      return setServerState(state, action.payload.serverID, {
        detailItems: {
          [action.payload.itemID]: action.payload.itemDetail
        }
      });
    },
    [actions.clearGeoEvents]: (state, action) => {
      return setServerState(state, action.payload.serverID, {
        geoEvents: []
      });
    },
    [actions.loadGeoEvents]: (state, action) => {
      return setServerState(state, action.payload.serverID, {
        geoEvents: [...action.payload.geoEvents]
      });
    }
  },
  {}
);
