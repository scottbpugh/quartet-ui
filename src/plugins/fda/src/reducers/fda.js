// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
import actions from "../actions/fda";
const {handleActions} = qu4rtet.require("redux-actions");
const {setServerState} = qu4rtet.require("./lib/reducer-helper");
const {showMessage} = qu4rtet.require("./lib/message");
const {pluginRegistry} = qu4rtet.require("./plugins/pluginRegistration");

export const initialData = () => {
  return {
    servers: {}
  };
};

export const searchFDA = (server, searchField, lookup, skip) => {
  return dispatch => {
    fetch(
      `https://api.fda.gov/drug/label.json?search=${searchField}:${lookup}&limit=20&skip=${skip}`
    )
      .then(response => {
        if (!response.ok) {
          throw Error(response);
        }
        return response.json();
      })
      .then(data => {
        if (data.results && data.meta) {
          return dispatch({
            type: actions.loadFDAResults,
            payload: {
              serverID: server.serverID,
              searchField: searchField,
              lookup: lookup,
              items: data.results,
              skip: data.meta.results.skip,
              total: data.meta.results.total
            }
          });
        }
      })
      .catch(error => {
        return dispatch({
          type: actions.loadFDAResults,
          payload: {
            serverID: server.serverID,
            searchField: searchField,
            lookup: lookup,
            items: [],
            skip: 0,
            total: 0
          }
        });
        console.log(error);
      });
  };
};

export default handleActions(
  {
    [actions.loadFDAResults]: (state, action) => {
      return setServerState(state, action.payload.serverID, {
        fdaSearchField: action.payload.searchField,
        fdaLookup: action.payload.lookup,
        fdaItems: action.payload.items,
        fdaSkip: action.payload.skip,
        fdaTotal: action.payload.total
      });
    }
  },
  {}
);
