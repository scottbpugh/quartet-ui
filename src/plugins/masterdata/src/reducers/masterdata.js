// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {handleActions} from "redux-actions";
import {pluginRegistry} from "plugins/pluginRegistration";
import actions from "../actions/masterdata";
import {showMessage} from "lib/message";

export const initialData = () => {
  return {
    servers: {}
  };
};

export const loadLocations = (server, search, page, ordering) => {
  let params = {};
  if (search) {
    params.search = search;
  }
  if (page) {
    params.page = page;
  }
  if (ordering) {
    params.ordering = ordering;
  }
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .fetchPageList("masterdata_locations_list", params, [])
      .then(response => {
        return dispatch({
          type: actions.loadLocations,
          payload: {
            serverID: server.serverID,
            locations: response.results,
            count: response.count,
            next: response.next
          }
        });
      })
      .catch(e => {
        showMessage({
          type: "error",
          id: "plugins.epcis.errorLoadingEvents",
          values: {error: e}
        });
      });
  };
};

export const loadCompanies = (server, search, page, ordering) => {
  let params = {};
  if (search) {
    params.search = search;
  }
  if (page) {
    params.page = page;
  }
  if (ordering) {
    params.ordering = ordering;
  }
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .fetchPageList("masterdata_companies_list", params, [])
      .then(response => {
        return dispatch({
          type: actions.loadCompanies,
          payload: {
            serverID: server.serverID,
            companies: response.results,
            count: response.count,
            next: response.next
          }
        });
      })
      .catch(e => {
        showMessage({
          type: "error",
          id: "plugins.epcis.errorLoadingCompanies",
          values: {error: e}
        });
      });
  };
};

export const loadTradeItems = (server, search, page, ordering) => {
  let params = {};
  if (search) {
    params.search = search;
  }
  if (page) {
    params.page = page;
  }
  if (ordering) {
    params.ordering = ordering;
  }
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .fetchPageList("masterdata_trade_items_list", params, [])
      .then(response => {
        return dispatch({
          type: actions.loadTradeItems,
          payload: {
            serverID: server.serverID,
            tradeItems: response.results,
            count: response.count,
            next: response.next
          }
        });
      })
      .catch(e => {
        showMessage({
          type: "error",
          id: "plugins.epcis.errorLoadingTradeItems",
          values: {error: e}
        });
      });
  };
};

export const deleteTradeItemField = (server, field) => {
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .getClient()
      .then(client => {
        client.apis.masterdata
          .masterdata_trade_item_fields_delete(field)
          .then(result => {
            return dispatch(loadTradeItems(server));
          });
      });
  };
};

export default handleActions(
  {
    [actions.loadLocations]: (state, action) => {
      if (!state.servers) {
        state.servers = {};
      }
      return {
        ...state,
        servers: {
          [action.payload.serverID]: {
            ...state.servers[action.payload.serverID],
            locations: action.payload.locations,
            count: action.payload.count,
            next: action.payload.next
          }
        }
      };
    },
    [actions.loadCompanies]: (state, action) => {
      if (!state.servers) {
        state.servers = {};
      }
      return {
        ...state,
        servers: {
          [action.payload.serverID]: {
            ...state.servers[action.payload.serverID],
            companies: action.payload.companies,
            count: action.payload.count,
            next: action.payload.next
          }
        }
      };
    },
    [actions.loadTradeItems]: (state, action) => {
      if (!state.servers) {
        state.servers = {};
      }
      return {
        ...state,
        servers: {
          [action.payload.serverID]: {
            ...state.servers[action.payload.serverID],
            tradeItems: action.payload.tradeItems,
            count: action.payload.count,
            next: action.payload.next
          }
        }
      };
    }
  },
  {}
);
