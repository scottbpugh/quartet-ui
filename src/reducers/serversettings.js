// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {handleActions} from "redux-actions";
import actions from "actions/serversettings";
import {showMessage} from "lib/message";
import uuidv4 from "uuid/v4";
import {pluginRegistry} from "plugins/pluginRegistration";
import {Server} from "lib/servers";

/**
 * initialData - Returns the initial data for
 * the Redux store for this form.
 * Used in index.js to initialized the store.
 *
 * It is a JSON-serializable object representing the form
 */
export const initialData = () => {
  let _initialData = {
    servers: {}, // uuid key to objects
    currentServer: null
  };
  return _initialData;
};

export const saveServer = postData => {
  return dispatch => {
    showMessage({type: "success", msg: "Your server settings were saved."});
    let server = {...postData};
    server.plugins = [];
    server.plugins.push("number-range");
    if (!postData.serverID) {
      postData.serverID = uuidv4();
    }
    dispatch({type: actions.saveServerSettings, payload: postData});
    const newServer = new Server(postData);
    pluginRegistry.registerServer(newServer);
    return dispatch({
      type: actions.serverUpdated,
      payload: JSON.stringify(newServer)
    });
  };
};

export const loadCurrentServer = serverData => {
  return dispatch => {
    return dispatch({type: actions.loadCurrentServer, payload: serverData});
  };
};

export default handleActions(
  {
    [actions.loadCurrentServer]: (state, action) => {
      return {
        ...state,
        formData: initialData(action.payload).formData
      };
    },
    [actions.saveServerSettings]: (state, action) => {
      return {
        ...state,
        servers: {...state.servers, [action.payload.serverID]: action.payload},
        formData: initialData().formData
      };
    },
    [actions.appsListUpdated]: (state, action) => {
      return {
        ...state,
        servers: {...state.servers, [action.payload.serverID]: action.payload}
      };
    }
  },
  {}
);
