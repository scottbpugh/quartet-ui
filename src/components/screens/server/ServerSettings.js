// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import React, {Component} from "react";
import PropTypes from "prop-types";
import {RightPanel} from "../../layouts/Panels";
import "./server-settings.css";
import {FormGroup, Switch, Card, Button} from "@blueprintjs/core";
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import {saveServer, loadCurrentServer} from "../../../reducers/serversettings";
import {FormattedMessage} from "react-intl";
import {DefaultField, getSyncValidators} from "components/elements/forms";
import {Field, reduxForm, change} from "redux-form";

const formStructure = (initialValues = {}) => [
  {
    name: "serverSettingName",
    description: {
      type: "text",
      required: true,
      read_only: false,
      label: "Server Setting Name",
      help_text:
        "The label that will be used for this server connection setting."
    }
  },
  {
    name: "serverID",
    description: {
      type: "hidden",
      required: false,
      read_only: false,
      label: "Server ID",
      help_text: "Hidden Server ID"
    }
  },
  {
    name: "serverName",
    description: {
      type: "text",
      required: true,
      read_only: false,
      label: "Server Hostname",
      help_text:
        "A hostname or IP address, example localhost, serial-box.com, or 192.168.5.10."
    }
  },
  {
    name: "port",
    description: {
      type: "number",
      required: true,
      read_only: false,
      label: "Port Number",
      min_value: 1,
      max_value: 65535,
      help_text: "A port to connect to. Example, 80, 8080, 443, ..."
    }
  },
  {
    name: "path",
    description: {
      type: "text",
      required: false,
      read_only: false,
      label: "Root Path",
      helperText: "A path to interact with API (Optional), example /api"
    }
  },
  {
    name: "ssl",
    description: {
      type: "boolean",
      required: false,
      read_only: false,
      label: "SSL/TLS",
      helperText: "SSL/TLS encryption"
    }
  },
  {
    name: "username",
    description: {
      type: "text",
      required: true,
      read_only: false,
      label: "Username",
      help_text: "Basic Auth Username"
    }
  },
  {
    name: "password",
    description: {
      type: "password",
      required: true,
      read_only: false,
      label: "Password",
      help_text: "Basic Auth Password"
    }
  }
];
/**
 * ServerForm - Description
 *
 * @param {Object} formData        structure representing input.
 * @param {function} saveHandler   callback to save the form
 * @param {function} updateHandler callback to update/validate each input
 *
 * @return {ReactElement} A form element with a series of inputs based on formData.
 */
class _ServerForm extends Component {
  componentDidMount() {
    this.formElems = this.props.formData.map(field => {
      field.validate = getSyncValidators(field);
      if (field.name === "serverSettingName") {
        field.validate.push(this.validateServerName.bind(this));
      }
      return (
        <Field
          key={field.name}
          name={field.name}
          fieldData={field}
          component={DefaultField}
          type={field.description.type}
          className="pt-input"
          width={300}
          validate={field.validate}
        />
      );
    });
  }
  submit = postValues => {
    this.props.saveServer(postValues);
  };
  validateServerName = value => {
    let serverSettingNames = Object.keys(this.props.servers).map(server => {
      return this.props.servers[server].serverSettingName.toLowerCase().trim();
    });
    if (serverSettingNames.includes(value.toLowerCase().trim())) {
      return "Server Setting Name already used.";
    }
    return undefined;
  };
  render() {
    const {handleSubmit, formData} = this.props;

    return (
      <form onSubmit={handleSubmit(this.submit)}>
        {this.formElems}
        <Button
          type="submit"
          className="pt-button pt-intent-primary"
          iconName="add">
          Add Server
        </Button>
      </form>
    );
  }
}
const ServerForm = connect(
  state => ({
    servers: state.serversettings.servers
  }),
  {saveServer, loadCurrentServer}
)(reduxForm({form: "serverForm"})(_ServerForm));

class _ServerSettings extends Component {
  constructor(props) {
    super(props);
    this.state = {postData: {}};
  }

  componentDidMount() {
    this.loadCurrentServer(this.props.match.params.serverID);
  }

  componentWillReceiveProps(nextProps) {
    this.loadCurrentServer(nextProps.match.params.serverID);
  }

  loadCurrentServer = serverID => {
    if (serverID) {
      if (serverID in this.props.servers) {
        this.setState({postData: {...this.props.servers[serverID]}}, () => {});
      }
    }
  };

  updateValue = evt => {
    // validation should happen here.
    if (evt.target.type === "checkbox") {
      // for checkboxes we passed checked prop instead.

      this.setState({
        postData: {
          ...this.state.postData,
          [evt.target.name]: evt.target.checked
        }
      });
      //this.props.updateValue(evt.target.name, evt.target.checked);
    } else {
      //this.props.updateValue(evt.target.name, evt.target.value);
      this.setState({
        postData: {...this.state.postData, [evt.target.name]: evt.target.value}
      });
    }
  };

  SettingsForm = props => {
    return (
      <div className="large-cards-container">
        <Card className="pt-elevation-4 form-card">
          <h5>Connect to a Server</h5>
          <ServerForm formData={formStructure()} />
        </Card>
      </div>
    );
  };

  SettingsMenu = props => {
    let serverList = Object.keys(props.servers).map(key => {
      return (
        <li key={props.servers[key].serverID}>
          <Link to={`/server-settings/${props.servers[key].serverID}`}>
            {props.servers[key].serverSettingName}
          </Link>
        </li>
      );
    });
    return <div />;
  };

  render() {
    return (
      <RightPanel
        title={
          <FormattedMessage
            id="app.serverSettings.serverSettings"
            defaultMessage="Server Settings"
          />
        }>
        {this.SettingsForm()}
      </RightPanel>
    );
  }
}

_ServerSettings.propTypes = {
  servers: PropTypes.object
};

export var ServerSettings = connect(
  state => ({
    servers: state.serversettings.servers
  }),
  {
    saveServer,
    loadCurrentServer
  }
)(_ServerSettings);
