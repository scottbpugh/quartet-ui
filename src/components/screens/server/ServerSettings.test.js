// Copyright (c) 2018 Serial Lab
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import React, {Component} from "react";
import renderer from "react-test-renderer";
import {ServerSettings} from "./ServerSettings";
import {initialData} from "../../../reducers/serversettings";
import configureStore from "redux-mock-store";
import {MemoryRouter as Router, withRouter} from "react-router-dom";
import {IntlProvider, intlReducer} from "react-intl-redux";

const mockStore = configureStore();
let wrapper;
let store;

it("renders correctly", () => {
  store = mockStore({serversettings: initialData()});
  const props = {
    match: {
      params: {
        serverID: null
      }
    }
  };
  const serverSettings = renderer
    .create(
      <IntlProvider store={store}>
        <Router>
          <ServerSettings {...props} store={store} />
        </Router>
      </IntlProvider>
    )
    .toJSON();
  expect(serverSettings).toMatchSnapshot();
});
