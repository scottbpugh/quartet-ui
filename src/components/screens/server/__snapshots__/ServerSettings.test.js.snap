// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="main-container"
>
  <div
    className="left-panel"
  >
    <h4
      className="pt-dark left-panel-title"
    >
      Server Settings
    </h4>
    <div>
      <ul />
      <a
        href="/server-settings/"
        onClick={[Function]}
      >
        <button>
          New Server
        </button>
      </a>
    </div>
  </div>
  <div
    className="right-panel"
  >
    <div
      className="settings-form-container"
    >
      <div
        className="pt-card pt-elevation-0"
        onClick={undefined}
      >
        <h5>
          Connect to a Server
        </h5>
        <form
          onSubmit={[Function]}
        >
          <div
            className="pt-form-group"
          >
            <label
              className="pt-label"
              htmlFor="serversetting-name"
            >
              Server Setting Name
            </label>
            <div
              className="pt-form-content"
            >
              <input
                className="pt-input"
                elemtype="input"
                id="serversetting-name"
                name="serverSettingName"
                onChange={[Function]}
                placeholder="Server/Connection Name"
                required={false}
                style={
                  Object {
                    "width": 300,
                  }
                }
                type="text"
                value=""
              />
              <div
                className="pt-form-helper-text"
              >
                The label that will be used for this server connection setting.
              </div>
            </div>
          </div>
          <div>
            <input
              className="pt-input"
              defaultValue=""
              elemtype="input"
              id="server-id"
              name="serverID"
              onChange={[Function]}
              required={false}
              style={
                Object {
                  "width": 300,
                }
              }
              type="hidden"
            />
          </div>
          <div
            className="pt-form-group"
          >
            <label
              className="pt-label"
              htmlFor="server-name"
            >
              Server Hostname
            </label>
            <div
              className="pt-form-content"
            >
              <input
                className="pt-input"
                elemtype="input"
                id="server-name"
                name="serverName"
                onChange={[Function]}
                placeholder="Server Hostname"
                required={true}
                style={
                  Object {
                    "width": 300,
                  }
                }
                type="text"
                value=""
              />
              <div
                className="pt-form-helper-text"
              >
                A hostname or IP address, example localhost, serial-box.com, or 192.168.5.10.
              </div>
            </div>
          </div>
          <div
            className="pt-form-group"
          >
            <label
              className="pt-label"
              htmlFor="port-number"
            >
              Port Number
            </label>
            <div
              className="pt-form-content"
            >
              <input
                className="pt-input"
                elemtype="input"
                id="port-number"
                max="65000"
                min="1"
                name="port"
                onChange={[Function]}
                placeholder="Port Number"
                required={true}
                style={
                  Object {
                    "width": 300,
                  }
                }
                type="number"
                value=""
              />
              <div
                className="pt-form-helper-text"
              >
                A port to connect to. Example, 80, 8080, 443, ...
              </div>
            </div>
          </div>
          <div
            className="pt-form-group"
          >
            <label
              className="pt-label"
              htmlFor="root-path"
            >
              Root Path
            </label>
            <div
              className="pt-form-content"
            >
              <input
                className="pt-input"
                elemtype="input"
                id="root-path"
                name="path"
                onChange={[Function]}
                placeholder="Root Path"
                required={false}
                style={
                  Object {
                    "width": 300,
                  }
                }
                type="text"
                value=""
              />
              <div
                className="pt-form-helper-text"
              >
                A path required to interact with API (Optional)
              </div>
            </div>
          </div>
          <div
            className="pt-form-group"
          >
            <label
              className="pt-label"
              htmlFor="ssl"
            >
              SSL/TLS
            </label>
            <div
              className="pt-form-content"
            >
              <label
                className="pt-control pt-switch"
                style={undefined}
              >
                <input
                  Label="SSL/TLS"
                  checked={undefined}
                  elemtype="Switch"
                  name="ssl"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="pt-control-indicator"
                />
              </label>
              <div
                className="pt-form-helper-text"
              >
                SSL/TLS encryption
              </div>
            </div>
          </div>
          <div
            className="pt-form-group"
          >
            <label
              className="pt-label"
              htmlFor="username"
            >
              Username
            </label>
            <div
              className="pt-form-content"
            >
              <input
                className="pt-input"
                elemtype="input"
                id="username"
                name="username"
                onChange={[Function]}
                placeholder="Username"
                required={true}
                style={
                  Object {
                    "width": 300,
                  }
                }
                type="text"
                value=""
              />
              <div
                className="pt-form-helper-text"
              >
                Basic Auth Username
              </div>
            </div>
          </div>
          <div
            className="pt-form-group"
          >
            <label
              className="pt-label"
              htmlFor="password"
            >
              Password
            </label>
            <div
              className="pt-form-content"
            >
              <input
                className="pt-input"
                elemtype="input"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                style={
                  Object {
                    "width": 300,
                  }
                }
                type="password"
                value=""
              />
              <div
                className="pt-form-helper-text"
              >
                Basic Auth Password
              </div>
            </div>
          </div>
          <button
            className="pt-button pt-icon-add"
            disabled={undefined}
            onClick={undefined}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="submit"
          >
            <span>
              Add Server
            </span>
          </button>
        </form>
      </div>
      <div
        className="pt-card pt-elevation-0"
        onClick={undefined}
      >
        <h5>
          Connection test & Server Info
        </h5>
      </div>
    </div>
  </div>
</div>
`;
